sonarqube-sample-java/src/main/java/com/consoleconnect/mw/sonarqube/sample/service/UserService.java

```diff
@@ -0,0 +1,40 @@
+/**
+ * 
+ */
+package com.consoleconnect.mw.sonarqube.sample.service;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+
+import com.consoleconnect.mw.sonarqube.sample.entity.User;
+
+/**
+ * @author dxiong
+ *
+ */
+public class UserService {
+
+	private Map<String, User> email2User = new HashMap<String, User>();
+
+	public User create(String email, String password) {
+		if (email2User.containsKey(email)) {
+			throw new IllegalArgumentException(String.format("%s already exist", email));
+		}
+
+		User user = new User(UUID.randomUUID().toString());
+		user.setEmail(email);
+		user.setPassword(password);
+
+		email2User.put(email, user);
+		
+		/////
+		/////
+		////
+		while(true) {
```

Comment:
![BLOCKER](https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER') Add an end condition to this loop. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/coding_rules#rule_key=squid%3AS2189)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either remove or fill this block of code. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/coding_rules#rule_key=squid%3AS00108)